kv/label.go

```diff
@@ -530,3 +617,41 @@ func (s *Service) deleteLabel(ctx context.Context, tx Tx, id influxdb.ID) error
 
 	return nil
 }
+
+// labelAlreadyExistsError is used when creating a new label with
+// a name that has already been used. Label names must be unique.
+func labelAlreadyExistsError(lbl *influxdb.Label) error {
+	return &influxdb.Error{
+		Code: influxdb.EConflict,
+		Msg:  fmt.Sprintf("label with name %s already exists", lbl.Name),
+	}
+}
+
+func labelIndexKey(l *influxdb.Label) ([]byte, error) {
+	orgID, err := l.OrgID.Encode()
+	if err != nil {
+		return nil, &influxdb.Error{
+			Code: influxdb.EInvalid,
+			Err:  err,
+		}
+	}
+
+	k := make([]byte, influxdb.IDLength+len(l.Name))
+	copy(k, orgID)
+	copy(k[influxdb.IDLength:], []byte(strings.ToLower((l.Name))))
```

Comment:
are we making names unique case insensitive now?

Reply:
Yup. label names should be unique & case agnostic. the UI currently handles all this stuff. feels like it's eagerly loading the labels and doing the label name validation in the browser...

for this change I added, the casing of the label names in the label bucket shouldn't matter & can be a mix of upper and lower case no problem. this [system bucket](https://github.com/influxdata/influxdb/pull/15647/files/6c7dccb1ceb67b80617cd7acc0b09855f9a24f83#diff-f1b793ee8b2acb0554850b94f082205cR17) is going to store the down-cased version of the label name, and we'll be using that to decide if a matching label name string already exists.